potato/server_utils/schemas/ranking.py

```diff
@@ -0,0 +1,158 @@
+"""
+ranking Layout
+"""
+
+import logging
+from collections.abc import Mapping
+
+logger = logging.getLogger(__name__)
+
+
+def generate_ranking_layout(annotation_scheme):
+    schematic = (
+            '<form action="/action_page.php">'
+            + "  <fieldset>"
+            + ("  <legend>%s</legend>" % annotation_scheme["description"])
+    )
+
+    # TODO: display keyboard shortcuts on the annotation page
+    key2label = {}
+    label2key = {}
+
+    key_bindings = []
+
+    display_info = (
+        annotation_scheme["display_config"] if "display_config" in annotation_scheme else {}
+    )
+
+    n_columns = display_info["num_columns"] if "num_columns" in display_info else 1
+
+
+    schematic += "<ul>"
+
+    # setting up label validation for each label, if "required" is True, the annotators will be asked to finish the current instance to proceed
+    validation = ""
+    label_requirement = (
+        annotation_scheme["label_requirement"] if "label_requirement" in annotation_scheme else None
+    )
+    if label_requirement and label_requirement["required"]:
+        validation = "required"
+
+    # if right_label is provided, the associated label has to be clicked to proceed. This is normally used for consent questions at the beginning of a survey.
+    right_label = set()
+    if label_requirement and "right_label" in label_requirement:
+        if type(label_requirement["right_label"]) == str:
+            right_label.add(label_requirement["right_label"])
+        elif type(label_requirement["right_label"]) == list:
+            right_label = set(label_requirement["right_label"])
+        else:
+            logger.warning("Incorrect format of right_label %s" % label_requirement["right_label"])
+            # quit()
```

Comment:
Do we have a "right label" in the click and drag? Could we delete this?

Reply:
the ranking module is still under testing, I'll remove this file in the next commit

Reply:
removed with [00f030e](https://github.com/davidjurgens/potato/pull/32/commits/00f030eed790a36423ba0da5ceda71682bc16319)